modules/store/src/selector.ts

```diff
@@ -390,44 +263,85 @@ export function createSelector<
     s5: S5,
     s6: S6,
     s7: S7,
+    s8: S8,
     props: Props
   ) => Result
 ): MemoizedSelectorWithProps<State, Props, Result>;
-export function createSelector<State, S1, S2, S3, S4, S5, S6, S7, Result>(
+
+export function createSelector<State, S extends unknown[], Result>(
+  selectors: [...{ [i in keyof S]: Selector<State, S[i]> }],
+  projector: (...s: S) => Result
+): MemoizedSelector<State, Result>;
+
+/**
+ * @deprecated Selectors with props are deprecated, for more info see {@link https://github.com/ngrx/platform/issues/2980 Github Issue}
+ */
+export function createSelector<State, Props, S1, Result>(
```

Comment:
left the existing definitions for `createSelector` when creating selectors with props since it's deprecated